### PR TITLE
refs[TASK][32827]: UI & Handle event Play Trailer Movie

### PR DESCRIPTION
**Pull request by Domhoan**
## Related Tickets
- [#Task 32827: UI & Handle event Play Trailer Movie](https://edu-redmine.sun-asterisk.vn/issues/32827)
## WHAT
- UI
image
image
- Handle: App will playing trailer movie when you click a movie
## HOW
- Get detail movie, then get url video of it, then put url into PlayYoutubeView and set properties for play screen
## WHY
- No
## Evidence (Screenshot or Video)
- The same UI
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/muvi/ui/trailer/TrailerFragment.kt [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            override fun onReady(@NonNull youTubePlayer: YouTubePlayer) {
+                youTubePlayer.loadVideo(youtubeId, 0f)
+                youtube = youTubePlayer
+            }
+        })
+
+        youtubeView.addFullScreenListener(object : YouTubePlayerFullScreenListener {
+            override fun onYouTubePlayerEnterFullScreen() {
+                requireActivity().requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
+                requireActivity().window.addFlags(WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS)
```
Comment (thonc-2135) [inline app/src/main/java/com/example/muvi/ui/trailer/TrailerFragment.kt]: `requireActivity()` vẫn null pointer exception đó

File: app/src/main/java/com/example/muvi/ui/trailer/TrailerFragment.kt [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        binding.youtubePlayerView.release()
+        bottomNavigationListener?.showNav()
+        super.onDestroy()
+    }
+
+    override fun initData() {
+        binding.lifecycleOwner = this
+        binding.detailViewModel = viewModel
+        binding.toolbarPlayTrailer.hide()
+        bottomNavigationListener?.hideNav()
```
Comment (thonc-2135) [inline app/src/main/java/com/example/muvi/ui/trailer/TrailerFragment.kt:74]: dùng scope function

File: app/src/main/java/com/example/muvi/ui/trailer/TrailerFragment.kt [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    override fun onPause() {
+        super.onPause()
+        youtube?.pause()
+    }
+
+    override fun onDestroy() {
+        binding.youtubePlayerView.release()
+        bottomNavigationListener?.showNav()
+        super.onDestroy()
+    }
```
Comment (thonc-2135) [inline app/src/main/java/com/example/muvi/ui/trailer/TrailerFragment.kt]: sắp xếp theo thứ tự vòng đời fragment

File: app/src/main/java/com/example/muvi/ui/detail/DetailViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+                    )
+                    .addTo(disposables)
+            } else {
+                favoriteRepository.insertFavorite(movieFavorite)
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe(
+                        {
+                            _isFavorite.value = true
+                            println(it)
```
Comment (thonc-2135) [inline app/src/main/java/com/example/muvi/ui/detail/DetailViewModel.kt]: lại quên xóa r

File: app/src/main/java/com/example/muvi/ui/detail/DetailViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+                    error.value = it.message
+                }
+            )
+            .addTo(disposables)
+    }
+
+    fun updateFavorite() {
+        detail.value?.let { it ->
+            val movieFavorite = Movie(it)
+            if (isFavorite.value == true) {
```
Comment (thonc-2135) [inline app/src/main/java/com/example/muvi/ui/detail/DetailViewModel.kt:125]:
```suggestion
            if (isFavorite.value) {
```

File: app/src/main/java/com/example/muvi/ui/trailer/TrailerFragment.kt (hunk 1 above)
Comment (Domhoan) [reply app/src/main/java/com/example/muvi/ui/trailer/TrailerFragment.kt]: ok a, thế phải check à a, hay có cách khác ạ

File: app/src/main/java/com/example/muvi/ui/detail/DetailViewModel.kt (hunk 5 above)
Comment (Domhoan) [reply app/src/main/java/com/example/muvi/ui/detail/DetailViewModel.kt:125]: cái isFavorite = LiveData<Boolean>, nên viết như kia là Boolean? ạ, mà e chỉ lấy true thôi, còn lại đẩy cho else

**Review by thonc-2135 — APPROVED** [review]

File: app/src/main/java/com/example/muvi/ui/trailer/TrailerFragment.kt (hunk 1 above)
Comment (thonc-2135) [reply app/src/main/java/com/example/muvi/ui/trailer/TrailerFragment.kt]: dùng activity? thôi

**Review by datdt-1935 — APPROVED** [review]

**Pull request merged**